lib/src/services/jira_api.ts

```diff
@@ -628,9 +625,10 @@ const jiraClient = (
       qualityReasonField: string,
       customFieldNames: readonly string[],
       descriptionFields: ReadonlyRecord<string, string>
+      // eslint-disable-next-line sonarjs/cognitive-complexity
```

Comment:
Rather than disabling the rule, a better way of handling the complexity would be to extract some of the code into its own function. 

This linting rule is especially helpful for future code-readers to understand what's happening in the code, so unless it's necessary to have super long/complicated functions, they're generally easier to comprehend when we split up the code into more digestible chunks. This also can help making it easier to test!